src/cirrus/lib/process_payload.py

```diff
@@ -277,14 +333,58 @@ def payload_ids(self) -> list[str]:
         """
         return [c["id"] for c in self.payloads]
 
-    def get_states(self) -> dict[str, StateEnum]:
+    def get_states_and_exec_arn(self) -> dict[str, tuple]:
         if self.state_items is None:
             items = [
                 self.statedb.dbitem_to_item(i)
                 for i in self.statedb.get_dbitems(self.payload_ids)
             ]
             self.state_items = items
-        return {c["payload_id"]: StateEnum(c["state"]) for c in self.state_items}
+        response = dict(self.get_process_attrs(self.state_items))
+        response.update(
+            {
+                p["id"]: (
+                    None,
+                    ProcessPayloads.gen_execution_arn(p["id"], p.process["workflow"]),
+                )
+                for p in self.payloads
+                if p["id"] not in response
+            },
+        )
+        return response
+
+    @staticmethod
+    def get_process_attrs(state_items):
+        for si in state_items:
+            payload_id = si["payload_id"]
+            state = StateEnum(si["state"]) if si["state"] is not None else None
+
+            # if workflow is found in CLAIMED state, then we want the last execution to
+            # determine where in the process of sfn execution the failure to proceeed to
+            # PROCESSING occurred.
+            if state == StateEnum.CLAIMED:
+                exec_arn = si["executions"][-1]
```

Comment:
I wouldn't hold up the PR because of this one line not being tested, but it does seem like if we could whip up a test hitting this condition it would be a good one to have covered.

Reply:
That shouldn't be too bad to add.